consensus/src/epoch_manager.rs

```diff
@@ -179,7 +180,41 @@ pub struct EpochManager<P: OnChainConfigProvider> {
     dag_config: DagConsensusConfig,
     payload_manager: Arc<PayloadManager>,
     rand_storage: Arc<dyn RandStorage<AugmentedData>>,
-    dkg_decrypt_key: <DefaultDKG as DKGTrait>::NewValidatorDecryptKey,
+    dkg_decrypt_key: Option<<DefaultDKG as DKGTrait>::NewValidatorDecryptKey>,
+}
+
+fn load_dkg_decrypt_key_from_identity_blob(node_config: &NodeConfig) -> anyhow::Result<<DefaultDKG as DKGTrait>::NewValidatorDecryptKey> {
+    let identity_blob = node_config.consensus.safety_rules.initial_safety_rules_config.identity_blob()?;
+    identity_blob.try_into_dkg_new_validator_decrypt_key()
+}
+
+fn load_dkg_decrypt_key_from_secure_storage(node_config: &NodeConfig) -> anyhow::Result<<DefaultDKG as DKGTrait>::NewValidatorDecryptKey> {
+    let storage: Storage = (&node_config.consensus.safety_rules.backend).try_into().map_err(|e|anyhow!("load_dkg_decrypt_key_from_storage failed with storage error"))?;
+    storage.available().map_err(|e|anyhow!("load_dkg_decrypt_key_from_secure_storage failed with storage unavailable"))?;
+    let sk = storage.get(CONSENSUS_KEY).map_err(|e|anyhow!("load_dkg_decrypt_key_from_secure_storage failed with storage read error: {e}"))?;
+    maybe_dk_from_bls_sk(&sk.value)
+}
+
+fn load_dkg_decrypt_key(node_config: &NodeConfig) -> Option<<DefaultDKG as DKGTrait>::NewValidatorDecryptKey> {
+    match load_dkg_decrypt_key_from_identity_blob(node_config) {
+        Ok(dk) => {
+            return Some(dk);
+        }
+        Err(e) => {
+            warn!("{e}");
+        }
+    }
+
+    match load_dkg_decrypt_key_from_secure_storage(node_config) {
```

Comment:
I think we can always load from secure storage, so don't need the step from identity blob?